### PR TITLE
Feature security

### PR DESCRIPTION
**Pull request by ya6lis**

### CODE REVIEW

**Comment by doliinyk** [comment]
PR should merge `feature-security -> develop`, not `main`

File: backend/controllers/authController.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        const { login, password } = req.body;
+
+        const data = await getUserDataForAuth(login);
+        const match = await bcrypt.compare(password, data.password);
+
+        if (match) {
+            const accessToken = jwt.sign(
+                { UserInfo: { login: data.login, is_admin: data.is_admin } },
+                process.env.ACCESS_TOKEN_SECRET,
+                { expiresIn: '30s' }
```
Comment (doliinyk) [inline backend/controllers/authController.js]: Access token should be 15-30m, 30s is so small

File: backend/controllers/refreshTokenController.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        jwt.verify(
+            refreshToken,
+            process.env.REFRESH_TOKEN_SECRET,
+            (error, decoded) => {
+                try {
+                    const accessToken = jwt.sign(
+                        { UserInfo: { login: decoded.login, is_admin: data.is_admin } },
+                        process.env.ACCESS_TOKEN_SECRET,
+                        { expiresIn: '30s' }
```
Comment (doliinyk) [inline backend/controllers/refreshTokenController.js]: It is better to make separate function in `authController` or `utils` and import here

File: backend/routes/api-refresh-routes.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const express = require('express');
+const router = express.Router();
+
+const refreshTokenController = require('../controllers/refreshTokenController');
+
+const url = '/api/v1/refresh';
+
+router.get(url, refreshTokenController.handleRefreshToken);
```
Comment (doliinyk) [inline backend/routes/api-refresh-routes.js:8]: Refresh should be POST, but it is not necessary and depends on meaning

File: backend/util/verifyRoles.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const verifyRoles = (allowedRole) => {
+    return (req, res, next) => {
+        if(!allowedRole === req.roles) return res.sendStatus(401);
```
Comment (doliinyk) [inline backend/util/verifyRoles.js]: `if (!allowedRole === req.roles)` means that user doesn't have enough permission to access resource, so it is `403`

**Review by doliinyk — CHANGES_REQUESTED** [review]

File: backend/controllers/refreshTokenController.js (hunk 2 above)
Comment (ya6lis) [reply backend/controllers/refreshTokenController.js]: They work differently. In authController - login: data.login and in refreshTokenController - login: decode.login. 
In authController we get it from dataBase and in refreshTokenController we get it from decoding refreshToken)

File: backend/util/verifyRoles.js [hunk 5]
```diff
@@ -1,8 +1,14 @@
+const {
+    getUserDataForAuth,
+} = require('../database/actionsWithTables/userActions');
+
 const verifyRoles = (allowedRole) => {
-    return (req, res, next) => {
-        if(!allowedRole === req.roles) return res.sendStatus(401);
+    return async (req, res, next) => {
+        const data = await getUserDataForAuth(req.user)
+        if(!data.is_admin === allowedRole){
```
Comment (doliinyk) [inline backend/util/verifyRoles.js]:
```suggestion
        if (!(data.is_admin === allowedRole)) {
```

**Review by doliinyk — APPROVED** [review]

**Pull request merged**